### PR TITLE
feat: PBI-2 Login

### PR DESCRIPTION
**Pull request by HugoIr**
### Describe your changes

- Add more login tests
- Integrated with login endpoint
- Handles authentication token

### Backlog name
[MON-42 Create login frontend](https://monapi.atlassian.net/browse/MON-42?atlOrigin=eyJpIjoiOWY0NDRiODU1ZTAwNDJhMWI4MWEzMGMwNDA5MjA5MGYiLCJwIjoiaiJ9)

### Acceptance Criteria

User able to login using their account
- User able to input email and password
- User receive error message when the input is not fulfill criteria

### Preview Page

Screen Shot 2022-09-30 at 09 50 24

### Checklist
- [x] I have performed a self-review of my code
- [x] Code successfully run on local
- [x] Feature / changes tested on local
- [x] No failing unit test
- [ ] Passed UAT Test
- [x] Sonarqube tested 
- [ ] Required any changes to environment variable

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +53,9 @@
     "setupFiles": [
       "<rootDir>/tests/__mocks__/browserMocks.js",
       "<rootDir>/tests/__mocks__/setupTests.js"
-    ]
+    ],
+    "transform": {
+      "^.+\\.(js|jsx)$": "babel-jest"
+    }
```
Comment (andrew-susanto) [inline package.json:60]: ini transformnya kayaknya gaperlu hug

File: src/components/app.js [hunk 2]
```diff
@@ -7,17 +7,20 @@ import theme from '../config/theme';
...
 			<ChakraProvider theme={theme}>
 				<Router>
-					<Home path="/" />
+					<Home path={ROUTE.DASHBOARD} />
 					<Login path={ROUTE.LOGIN} />
+					<div path={ROUTE.REGISTER}></div>
 				</Router>
 			</ChakraProvider>
 		</Provider>
+		<AuthenticationChecker/>
```
Comment (andrew-susanto) [inline src/components/app.js]:
```
	          <Provider store={store}>
			<ChakraProvider theme={theme}>
				<Router>
					<Home path="/" />
                                        <Login path={ROUTE.LOGIN} />
					<div path={ROUTE.REGISTER}></div>
		                        <AuthenticationChecker path="/:*?">
					          <Home path={ROUTE.DASHBOARD} />
		                        </AuthenticationChecker>
				</Router>
			</ChakraProvider>
		</Provider>

```
buat authenticated nya mending dibuat jadi wrapper aja gak hug?

nanti di authenticatednya kalo dia authenticated render children nya

File: package.json (hunk 1 above)
Comment (HugoIr) [reply package.json:60]: okayy

File: src/components/app.js (hunk 2 above)
Comment (HugoIr) [reply src/components/app.js]: i see ternyata harus ditambahin path="/:*?", okee thanks drew

**Review by andrew-susanto — APPROVED** [review]

**Review by luthfifahlevi — APPROVED** [review]
Bagus, coverage jg udh 100%, sdh deskriptif jg codenya

**Pull request merged**